File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size() - 1;
        int maxVol = -1;
        int vol;
        while (i < j)
        {
            if (height[i] < height[j])
            {
                vol = (j - i) * height[i];
                if (vol > maxVol)
                {
                    maxVol = vol;
                }
                i ++;
            }
            else
            {
                vol = (j - i) * height[j];
                if (vol > maxVol)
                {
                    maxVol = vol;
                }
                j --;
            }
        }
        return maxVol;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0,h=height.length-1;
        int mw=0;
        while(l<h){
            int sh=Math.min(height[l],height[h]);
            int cm=(h-l)*sh;
            mw=Math.max(mw,cm);
            if(height[h]>height[l]){
                l++;
            }
            else{
                h--;
            }
            
        }
        return mw;
    }
}